Fix #16892: Chromevox doesn't read anything when focus is on headphones or expand icons in Listen to the lesson bar

rabbanimdrb opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #16892.
2. This PR does the following: **It makes the audio component in exploration interactive**.

## Essential Checklist

- [x] The PR title starts with "Fix #16892: ", The PR made the **headphone**, **expand**, **play-button** **icons** focusable and added **desctiptive aria-label attributes** to the respective elements so the interactive element's functionality can  be understood easily,and **added the functionality to **use enter button** to access the interactive elements**,  and it also **added focus on the tutor-card's top component which was missing**

<!--
followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #16892: ...".)
-->
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. 
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct


https://user-images.githubusercontent.com/69578560/221532402-79a6a58b-bee7-47dc-9534-4fb8fb4696e4.mp4


<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network - **N/A**

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone - **N/A**

<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language - **N/A**

<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

rabbanimdrb committed:
Added functionality to focus on the headphone, expand, play-button icons and desctiptive aria-label attributes so the interactive element's functionality can  be understood easily

rabbanimdrb committed:
Added focus on the tutor-card's top component

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb commented:
Adding video of the changes working soon.

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb commented:
@Lawful2002 All checks have passed successfully.Could you please take a review.

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

Lawful2002 commented:
@SanjaySajuJacob Can you please take a look as a oppia/lace-frontend-reviewer

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

SanjaySajuJacob reviewed (commented):
Hey @rabbanimdrb , the changes look fine to me but would you just show the proof that this particular dropdown is also accessible on pressing enter.

![image](https://user-images.githubusercontent.com/70763993/222853900-ccf97a85-5b67-4720-b8dc-dcfff83f28be.png)

If not can that be done. Thanks!

rabbanimdrb commented:
> Hey @rabbanimdrb , the changes look fine to me but would you just show the proof that this particular dropdown is also accessible on pressing enter.
> 
> ![image](https://user-images.githubusercontent.com/70763993/222853900-ccf97a85-5b67-4720-b8dc-dcfff83f28be.png)
> 
> If not can that be done. Thanks!

I will see to it, if it is not accesible , I will add that functionality .

Thanks @SanjaySajuJacob

SanjaySajuJacob commented:
Great! Thanks @rabbanimdrb

rabbanimdrb commented:
@SanjaySajuJacob I just check that the language dropdown menu is accesible but it's not assoicated with the **enter** key rather it can be accessed by the arrow up or down button.

Is it fine or I shall go ahead and change anything to it, I have even attached a video so that you can see it yourself how it actually works.

![image](https://user-images.githubusercontent.com/69578560/222968194-da27a44c-aed7-4a13-b484-978ce97b1c39.png)


https://user-images.githubusercontent.com/69578560/222976590-f15ae587-2108-44a3-91d4-f73a6b011e95.mp4

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb commented:
> @SanjaySajuJacob I just check that the language dropdown menu is accesible but it's not assoicated with the **enter** key rather it can be accessed by the arrow up or down button.
> 
> Is it fine or I shall go ahead and change anything to it, I have even attached a video so that you can see it yourself how it actually works.
> 
> ![image](https://user-images.githubusercontent.com/69578560/222968194-da27a44c-aed7-4a13-b484-978ce97b1c39.png)

@SanjaySajuJacob Could you please take a look.

rabbanimdrb commented:
> Unassigning @rabbanimdrb since a re-review was requested. @rabbanimdrb, please make sure you have addressed all review comments. Thanks!

Yes I have addressed all the reviews.

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

rabbanimdrb committed:
Merge branch 'develop' into no-focus-is-on-headphones-or-expand-icons

Lawful2002 reviewed (approved):
LGTM Thanks!
